tools/make_class_power.py

```diff
@@ -0,0 +1,206 @@
+Sigma8 = float(default=-1)
+InputPowerRedshift = float()
+InputFutureRedshift = float()
+DifferentTransferFunctions = integer(0,1)
+InputSpectrum_UnitLength_in_cm  = float(default=3.085678e24)
+UnitLength_in_cm  = float(default=3.085678e21)
+Omega_fld = float(0,1,default=0)
+w0_fld = float(default=-1)
+wa_fld = float(default=0)
+MNue = float(min=0, default=0)
+MNum = float(min=0, default=0)
+MNut = float(min=0, default=0)
+MWDM_Therm = float(min=0, default=0)
+PrimordialIndex = float(default=0.971)
+PrimordialAmp = float(default=2.215e-9)
+CMBTemperature = float(default=2.7255)""".split('\n')
+
+def _check_genic_config(config):
+    """Check that the MP-GenIC config file is sensible for running CLASS on."""
+    vtor = validate.Validator()
+    config.validate(vtor)
+    filekeys = ['FileWithInputSpectrum', ]
+    if config['DifferentTransferFunctions'] == 1.:
+        filekeys += ['FileWithTransferFunction', 'FileWithFutureTransferFunction']
+    for ff in filekeys:
+        if config[ff] == '':
+            raise IOError("No savefile specified for ",ff)
+        if os.path.exists(config[ff]):
+            raise IOError("Refusing to write to existing file: ",config[ff])
+
+    #Make sure MP-GenIC expects input in Mpc/h!
+    iinMpc = config['InputSpectrum_UnitLength_in_cm']/3.085678e24
+    if abs(iinMpc-1) > 1e-6:
+        raise AssertionError("CLASS outputs power spectrum in Mpc/h units, MP-GenIC expects %.5f Mpc/h." % iinMpc)
+    #Check unsupported configurations
+    if config['MWDM_Therm'] > 0:
+        raise ValueError("Warm dark matter power spectrum cutoff not yet supported.")
+    if config['DifferentTransferFunctions'] == 1.:
+        if config['InputPowerRedshift'] >= 0:
+            raise ValueError("Rescaling with different transfer functions not supported.")
+
+def _build_cosmology_params(config):
+    """Build a correctly-named-for-class set of cosmology parameters from the MP-GenIC config file."""
+    #Class takes omega_m h^2 as parameters
+    h0 = config['HubbleParam']
+    omeganu = (config['MNue'] + config['MNum'] + config['MNut'])/93.14/h0**2
+    ocdm = config['Omega0'] - config['OmegaBaryon'] - omeganu
+    omegak = 1-config['OmegaLambda']-config['Omega0']
+    gparams = {'h':config['HubbleParam'], 'Omega_cdm':ocdm,'Omega_b':config['OmegaBaryon'], 'Omega_k':omegak, 'w0_fld': config['w0_fld'], 'wa_fld':config['wa_fld'], 'n_s': config['PrimordialIndex'],'T_cmb':config["CMBTemperature"]}
+    if config['Omega_fld'] > 0:
+        gparams['Omega_fld'] = config['Omega_fld']
+    else:
+        gparams['Omega_Lambda'] = config['OmegaLambda']
+    #Set up massive neutrinos
+    if omeganu > 0:
+        gparams['m_ncdm'] = '%.2f,%.2f,%.2f' % config['MNue'], config['MNum'], config['MNut']
+        gparams['N_ncdm'] = 3
+    #Power spectrum amplitude
+    if config['Sigma8'] > 0:
+        gparams['sigma8'] = config['Sigma8']
+    else:
+        #Pivot scale is by default 0.05 1/Mpc! This number is NOT what is reported by Planck.
```

Comment:
There is this comment but no action -- did you want to set something here?

Reply:
I was unsure - in many ways a pivot scale of 0.05 1/Mpc is sensible for large scale structure, but it does mean that the A_s values are different from what is in the Planck papers, where pivot is 2e-3. If we want to be able to use Planck parameters we should change it. I wasn't sure which is more obvious.

Reply:
I wasn't aware if a different pivot scale is not sensible for large scale structure, is it? 

I pulled the number nick put in nbodykit for Planck15, and it is larger than 0.05. The number (0.07ish?) looked somewhat strange does it?

```
In [1]: from nbodykit.cosmology import Planck15

In [2]: Planck15.k_pivot
Out[2]: 0.07381163271331562

In [3]: Planck15.A_s
Out[3]: 2.1413038238853928e-09
```